rtmp-client/src/main/cpp/librtmp/rtmp.c

```diff
@@ -1037,28 +1028,26 @@ RTMP_Connect1(RTMP *r, RTMPPacket *cp)
     {
       RTMP_Log(RTMP_LOGERROR, "%s, handshake failed.", __FUNCTION__);
       RTMP_Close(r);
-      r->m_error = RTMP_ERROR_HANDSHAKE_FAIL;
-      return FALSE;
+      return RTMP_ERROR_HANDSHAKE_FAIL;
     }
   RTMP_Log(RTMP_LOGDEBUG, "%s, handshaked", __FUNCTION__);
 
-  if (!SendConnectPacket(r, cp))
+  if (SendConnectPacket(r, cp) != RTMP_SUCCESS)
```

Comment:
I think here it might be better to send `RTMP_ERROR_CONNECT_FAIL` as the error `RTMP_ERROR_SEND_PACKET_FAILED` could have been returned from any call to `RTMP_SendPacket` method. `RTMP_SendPacket` is called from other subroutines within `nativeOpen`, like `RTMP_Connect0`.

I feel it better points to the cause of the error. Please let me know your thoughts.

Reply:
That makes sense; thanks!